### DISCUSSION TITLE
service unavailable: CalcDistance deprecated

### DISCUSSION THREAD

**Discussion by littlePoBoy** [milvus-io/milvus#35481] [Q&A and General discussion] [answered]
when i calculate distance between two vectors with sdk-go CalcDistance() function,An error has occurred：service unavailable: CalcDistance deprecated。
i use **milvus-sdk-go version v2.4.1。**
**milvus standalone version v2.4.6**
`
        vector1 := entity.NewColumnFloatVector("vector", 512, v1)
	vector2 := entity.NewColumnFloatVector("vector", 512, v2)

	distance, err := mc.CalcDistance(ctx, param.ColName, []string{}, entity.L2, vector1, vector2)
	if err != nil {
		slog.Error("calc distance error", slog.Any("err", err))
		return 0, err
	}`

if this function CalcDistance() is deprecated ，How to calculate the distance between two vectors with milvus-sdk-go?
Is there any other way to calculate distance between two vectors?
thanks!!!

**Comment by yhmo** [comment, marked as answer]
The calcDistance interface has been removed from the milvus server side. Most of the SDK have removed this interface except the Go SDK, we will remove it later.

You can use python numpy to calculate distance between vectors easily:
https://www.geeksforgeeks.org/calculate-the-euclidean-distance-using-numpy/

**Reply by littlePoBoy** [reply]
thanks your reply，Could you please let me know which version was removed from

**Reply by yhmo** [reply]
Removed from v2.3.0 by this pr: https://github.com/milvus-io/milvus/pull/25663
